src/components/Pagination.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from "react";
+
+class Pagination extends Component {
+
+    render() {
+        const currentPage = this.props.currentPage;
+        const numPages = this.props.numPages;
+        // subset of page numbers to display
+        let pageNumbers = [...new Set([
+            1, 2, 3,
+            Math.max(currentPage-1, 1), currentPage, Math.min(currentPage+1, numPages),
+            numPages-2, numPages-1, numPages
+        ])].sort((a, b) => a - b);
```

Comment:
This assume you'll have at least 3 pages. If you only have one page that won't work.